seed/challenges/01-front-end-development-certification/json-apis-and-ajax.json

```diff
@@ -340,7 +340,7 @@
         "</div>"
       ],
       "tests": [
-        "assert(code.match(/val.imageLink/gi), 'message: You should have used the <code>imageLink</code> property to display the images.');"
+        "assert(code.match(/(html) *(\\+=) *\\\"<{1}(i|I)(m|M)(g|G) *(s|S)(r|R)(c|C) *= *'{0,1}\" *\\+ *(val.imageLink) *\\+ *\" *'{0,1} *>\";/g), 'message: You should have used the <code>imageLink</code> property to display the images.');"
```

Comment:
This is an overly complex regex.  
You should never need `{1}` because simply putting a character means one and only one of a character.

There is no need for all of the capture groups to get upper and lower case letter, instead just use case insensitive flag (`/ig`).

`{0,1}` can be replaced with `?` (which means zero or one of the preceding character).

Additionally, I see ` *` (Space star) which is probably more properly `\s*` to allow for tabs or even line-breaks.

I'm also not sure why there are capture groups around some internal strings, like `(val.imageLink)` - also the `.` in that string should be escaped.